types/react/index.d.ts

```diff
@@ -2788,11 +2788,13 @@ declare namespace React {
     // ----------------------------------------------------------------------
 
     interface ReactChildren {
-        map<T, C>(children: C | C[], fn: (child: C, index: number) => T): T[];
+        map(children: null, fn: (child: never, index: number) => never): null;
+        map(children: undefined, fn: (child: never, index: number) => never): undefined;
+        map<T, C>(children: C | C[], fn: (child: C, index: number) => T): Array<Exclude<T, boolean | null | undefined>>;
```

Comment:
Never consciously thought about this :+1: I'm actually a bit scared now considering I always thought this works just like Array.prototype.map.

I feel like the documentation should definitely mention that the result is filtered. I would expect the length of the array to stay the same when mapping i.e. `React.Children.count(children) === React.Children.count(React.Children.map(children, c => c))` should always be `true`.

Reply:
> I feel like the documentation should definitely mention that the result is filtered.

I completely agree. The docs don't describe the full situation :-/